src/plugins/intel_gpu/src/plugin/transformations/decompose_reduce_scalar_output.cpp

```diff
@@ -34,6 +34,8 @@
 ov::intel_gpu::DecomposeReduceForScalarOutput::DecomposeReduceForScalarOutput() {
     auto check_reduce_shape = [=](Output<Node> output) -> bool {
         const auto reduce = ov::as_type_ptr<op::util::ArithmeticReductionKeepDims>(output.get_node_shared_ptr());
+        if (reduce == nullptr)
+            return false;
```

Comment:
I think this can be better expressed with `assert` because it is not supposed to be `nullptr`